1-Draft/RFCXXXX-Obsoletion-and-Deprecation.md

```diff
@@ -0,0 +1,511 @@
+        Message = "This cmdlet does not guarantee secure connections to SMTP servers. While there is no immediate replacement available in PowerShell, we recommend you do not use Send-MailMessage at this time. See https://aka.ms/SendMailMessage for more information."
+        Action = "NotFound"
+      }
+    }
+  }
+}
+...
+```
+
+- the MemberSignature is for disambiguation of obsolescence for the member of a type which may have overloaded signatures.
+- If there are no obsolete types, then `Types` may be omitted.
+- If there are no obsolete commands (cmdlets, functions, or scripts), then `Commands` may be omitted.
+- If there are no obsolete parameters, then `Parameters` may be omitted.
+- If there are no obsolete members, then `Members` may be omitted.
+
+The actual `ObsoleteAttribute` must still be present in the command or type.
+The source of truth once the module is imported is the presence of the `ObsoleteAttribute` in the source of the cmdlet, parameter, type, or member.
+The elements are present in the manifest to not require that the module is loaded before being able to report on obsolescence.
+
+## Configuration Tools
+
+There are a number of elements that require configuration for managing the behavior when an element is marked for obsolescence.
+Because of this, we shall provide cmdlets to aid with the creation of the configuration.
+Three cmdlets, `Get-ObsoleteItem`, `Enable-ObsoleteItem`, and `Disable-ObsoleteItem` shall be provided.
+
+### `Get-ObsoleteItem`
+
+  This cmdlet retrieves those items which have been attributed as obsolete and their current configuration.
+  By default this will return only cmdlets whose items which are currently in loaded modules.
+  `-All` shall check all available modules and inspect the PrivateData element for obsolete items and report them, if found.
+  `-Force` shall inspect all loaded types and members in PowerShell assemblies for the obsolete attribute and report them, if found. Note that this may take some time and should be used rarely.
+  `-ItemType` allows you to retrieve a partial list based on the ItemType.
+  Allowed values are `Type`, `Command`, `ExternalScript`
+
+  The output of the `Get-ObsoleteItem` has at least the following properties:
+
+```cs
+enum ItemType {
+    Type
+    Command
+    ExternalScript
+}
+
+enum MemberType {
+    Parameter
+    Property
+    Field
+    Method
+    Type
+    Event
+    Constructor
+}
+
+class ObsoleteMember {
+    [string]$Name
+    [string]$Signature
+    [string]$Message
+    [bool]$IsError
+    [string]$WarningId
+    [MemberType]$MemberType
+}
+
+class ObsoleteItem {
+    [string]$FullyQualifiedWarningId
+    [string]$Message
+    [bool]$IsError
+    [string]$Module
+    [ItemType]$ItemType
+    [bool]$IsObsolete
+    [string]$WarningId
+    [bool]$HasObsoleteMembers
+    [List[ObsoleteMember]]$ObsoleteMembers
+}
+```
+
+The FullyQualifiedWarningId is a calculated property made up of the Name of the type or the cmdlet
```

Comment:
Shall we use the new property `DiagnosticId`, which was newly added to `ObsoleteAttribute`?